l2l/optimizers/kalmanfilter/enkf.py

```diff
@@ -0,0 +1,155 @@
+import numpy as np
+import abc
+
+from .kalman_utils import _get_shapes, _encode_targets, _get_batches
+from abc import ABCMeta
+
+try:
+    from numba import jit
+except ModuleNotFoundError:
+    # create a mock decorator if `numba` is not found
+    def jit(**kwargs):
+        def decorator(f):
+            def wrapper(*args, **kwargs):
+                return f(*args, **kwargs)
+            return wrapper
+        return decorator
+
+
+class KalmanFilter(metaclass=ABCMeta):
```

Comment:
Since there's only one concrete instance of this class, does it make sense to have this abstract class at all?

Reply:
I had extensibility in mind. Because the Ensemble Kalman Filter is just one version of the Kalman Filter. So I did a meta class. In my case a have several implementations which profit from the abstract class design.  
I can change it though.

Reply:
I would suggest putting in an abstract class *when* you actually extend it :-)